test-data/unit/check-inference.test

```diff
@@ -856,13 +856,41 @@ s = s_s() # E: Incompatible types in assignment (expression has type "Set[str]",
 [builtins fixtures/set.pyi]
 
 [case testSetWithStarExpr]
-
 s = {1, 2, *(3, 4)}
 t = {1, 2, *s}
 reveal_type(s)  # E: Revealed type is 'builtins.set[builtins.int*]'
 reveal_type(t)  # E: Revealed type is 'builtins.set[builtins.int*]'
 [builtins fixtures/set.pyi]
 
+[case testListLiteralWithFunctionsErasesNames]
```

Comment:
Maybe add a test for the `join_similar_callables` path by joining functions whose argument types differ but have a nontrivial meet. (For example, if the arguments are `Union[bytes, float]` and `Union[str, int]`, the argument for the joined function should be `int`)